spec/db/writer_spec.rb

```diff
@@ -9,12 +9,27 @@
   ]}
 
   let(:contents) {'[{"name":"name1","address":"address1"},{"name":"name2","address":"address2"},{"name":"name3","address":"address3"}]'}
-  let (:file)    {instance_double(File).as_null_object}
-  let (:writer)  {described_class.new("", file)}
 
-  it "saves contacts to a file" do
-    writer.run(contacts)
-    expect(file).to have_received(:write).with(contents)
+  it "replaces the contents of the file" do
+    file        = StringIO.new(contents)
+    one_contact = [{"name"=>"name1", "address"=>"address1"}]
+    writer      = described_class.new(file)
+    writer.write(one_contact)
+    expect(file.string).to eq('[{"name":"name1","address":"address1"}]')
+  end
+
+  it "saves the contents of the file in JSON format" do
```

Comment:
nothing in the body of this test gives me a clue as to how you're verifying that the file is json. guessing contents is json, but that is defined far away from here. perhaps the test just needs a better name, as it is i expect some assertion about json.